### ISSUE TITLE
[BUG]: Custom models can't extend Pivot relation

### ISSUE THREAD

**Issue by maxrice** [dimitriBouteille/wp-orm#75]
### Describe the bug

When creating a many-to-many relationship, Eloquent allows you to [define a custom model](https://laravel.com/docs/11.x/eloquent-relationships#defining-custom-intermediate-table-models) that will be used for the intermediate table. To do this, the model must extend the [`Pivot` relation](https://github.com/illuminate/database/blob/master/Eloquent/Relations/Pivot.php). That model is a simple wrapper over the base `Model` class, with a [`AsPivot` trait](https://github.com/illuminate/database/blob/master/Eloquent/Relations/Concerns/AsPivot.php).

To do this using the WP ORM library, I've extended the base [`AbstractModel` class](https://github.com/dimitriBouteille/wp-orm/blob/develop/src/Orm/AbstractModel.php) and added `use AsPivot`. However, this triggers an error:

```
Declaration of Illuminate\Database\Eloquent\Relations\Concerns\AsPivot::getTable() must be compatible with  Dbout\WpOrm\Orm\AbstractModel::getTable(): string
```

The Eloquent base `Model` class does not add a return type to `getTable()`, so it seems that in order to fully support many-to-many relationships, you'll need to remove the `string` return type for `getTable()`](https://github.com/dimitriBouteille/wp-orm/blob/develop/src/Orm/AbstractModel.php#L36-L45)

### Your setup

- WordPress Version: 6.6.1
- Module version: 3.2.0
- Are you using framework ?: no

**Comment by dimitriBouteille** [comment]
Hi @maxrice 

Thanks for the details. I look to fix this bug as quickly as possible, you will be informed when it is done.

Best,

**Comment by dimitriBouteille** [comment]
Hey @maxrice 

The bug has been fixed in [v3.3.0](https://github.com/dimitriBouteille/wp-orm/releases/tag/3.3.0) :rocket:

**Comment by maxrice** [comment]
wonderful, thanks!
